forge/src/swarm/SwarmV2.sol

```diff
@@ -0,0 +1,422 @@
+        (penalty, finderReward) = _slashTrainerCommitment(roundId, commitment, msg.sender);
+        if (finderReward > 0) {
+            (bool success, ) = payable(msg.sender).call{value: finderReward}("");
+            if (!success) {
+                revert TransferFailed(msg.sender, finderReward);
+            }
+        }
+    }
+
+    function configureTrainerPrivacy(uint256 penalty, uint16 finderRewardBps) external onlyAggregator(msg.sender) {
+        _setTrainerPrivacyConfig(penalty, finderRewardBps);
+    }
+
+    function setTrainerPrivacyMode(bool enabled) external onlyAggregator(msg.sender) {
+        if (_trainerPrivacyEnabled == enabled) {
+            return;
+        }
+        _trainerPrivacyEnabled = enabled;
+        emit TrainerPrivacyModeUpdated(enabled);
+    }
+
+    function depositAggregatorBond() external payable onlyAggregator(msg.sender) {
+        if (msg.value == 0) {
+            revert ZeroBondAmount();
+        }
+        _increaseAggregatorBond(msg.value);
+        (uint256 balance, ) = getAggregatorBondState();
+        emit AggregatorBondDeposited(msg.sender, msg.value, balance);
+    }
+
+    function withdrawAggregatorBond(uint256 amount, address payable recipient) external onlyAggregator(msg.sender) {
+        if (amount == 0) {
+            revert ZeroBondAmount();
+        }
+        if (recipient == address(0)) {
+            revert InvalidRecipient();
+        }
+        _decreaseAggregatorBond(amount);
+        (bool success, ) = recipient.call{value: amount}("");
+        if (!success) {
+            revert TransferFailed(recipient, amount);
+        }
+        (uint256 balance, ) = getAggregatorBondState();
+        emit AggregatorBondWithdrawn(msg.sender, recipient, amount, balance);
+    }
+
+    function getAggregatorFreeBond() external view returns (uint256) {
+        (uint256 balance, uint256 reserved) = getAggregatorBondState();
+        return reserved >= balance ? 0 : balance - reserved;
+    }
+
+    function isTrainerPrivacyEnabled() external view returns (bool) {
+        return _trainerPrivacyEnabled;
+    }
+
+    function startTrainingRound() external onlyAggregator(msg.sender) {
+        if (updatePhase() != IDLE_PHASE) {
+            revert NotIdle();
+        }
+        _nextRound();
+        _forceStartTrainingPhase();
+    }
+
+    function _endTrainingPhase() internal override returns (bytes32) {
+        uint256 numberOfTrainers = getTrainerCount(currentRound());
+        uint256 pendingCommitments = getPendingCommitmentCount(currentRound());
+        if (pendingCommitments > 0) {
+            numberOfTrainers += pendingCommitments;
+        }
+        if (numberOfTrainers <= 0) {
+            return TRAINING_PHASE;
+        }
+        return super._endTrainingPhase();
+    }
+
+    function registerRoundContribution(uint256 roundId, bytes32 modelHash) external {
+        if (_trainerPrivacyEnabled) {
+            revert PrivacyModeEnabled();
+        }
```

Comment:
OK, so the boolean switch is for backward compatibility, but I think it would be cleaner getting rid of this function and other functionalities which aren't required for your specific implementation